### PR TITLE
Jazzed up history page

### PR DESCRIPTION
**Pull request by samozzy**
- Minimise blocks of text into year-based events
- Move all events into a datafile (_data/history.yaml)
- New layout to present this data (_layouts/history.html)

(See #474)

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
Looks good! You seem to have reinstated the #fakeNews about lakeside though

**Comment by samozzy** [comment]
Don't think so - the #fakenews was the idea that the first New Theatre show at the Djanogly Theatre was the first Lakeside collab (c.2003), when it's in fact Lysistrata in 2013. Do correct me if I'm wrong though!

**Comment by davidtaylorhq** [comment]
Ah I see, that makes sense. I read 2003 as 2013

File: _content/history.md [hunk 1]
```diff
@@ -1,43 +1,14 @@
 ---
-layout: page
+layout: history
 title: History
 group: top
 current: history
 ---
 
-# History of The Nottingham New Theatre
+# What is The Nottingham New Theatre?
```
Comment (davidtaylorhq) [inline _content/history.md]: I'm not sure a title is needed here - it takes up a lot of space. Going straight in with "The nottingham new theatre is..." is pretty self explanatory.

File: _data/history.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Chronological history of the Nottingham New Theatre
+
+- year: 1927
+  title: Formation of Dramsoc
+  description: "The University College Nottingham Dramatic Society (Dramsoc) performed its first play in 1928: A.A. Milne's *Mr Pim Passes By*. <br /> The society offered members an opportunity to read, discuss, produce and enact plays."
```
Comment (davidtaylorhq) [inline _data/history.yaml]: For all the description fields, I dislike the super long lines with <br /> in them. Can YML have multi-line fields?

File: _layouts/history.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+layout: page
+body_class: layout-page
+---
+
+{{ content }}
+<h1> History of the Nottingham New Theatre </h1>
```
Comment (davidtaylorhq) [inline _layouts/history.html]: See above, I don't think the massive titles are strictly necessary. Going straight in with "1927 - Formation of Dramsoc" is pretty self explanatory IMO

File: _layouts/history.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+{% for node in site.data.history %}
+	<h2>{{ node.year }} - {{ node.title }}</h2>
+	{% if node.img %} <!-- Images! -->
+		<img src="{{ node.img }}" {% if node.alt-img %} alt="{{ node.alt-img }}" {% endif %} class="page-photo {% cycle 'right','left' %}"/>
```
Comment (davidtaylorhq) [inline _layouts/history.html]: I expect @wjdp will want to smugmug-ify the images at some point, but this'll work for now :)

File: _data/history.yaml (hunk 2 above)
Comment (wjdp) [inline _data/history.yaml]: It can, google it

Comment (samozzy) [inline _data/history.yaml]: I agree with you. I think they must get stripped as the line breaks don't get processed in markdownify process.

Comment (davidtaylorhq) [inline _data/history.yaml]: Might need double line breaks, because markdown?

Comment (wjdp) [inline _data/history.yaml]: Single BRs in paragraph text are not what we want. You could do multiline yaml then use the `| markdownify` filter to get paragraph tags.

Multiline strings: https://stackoverflow.com/questions/3790454/in-yaml-how-do-i-break-a-string-over-multiple-lines

Disclaimer:

> There are ~~5 6~~ NINE different ways to write multi-line strings in YAML

I've used the `>` method in the past.

File: _content/history.md (hunk 1 above)
Comment (wjdp) [inline _content/history.md]: Agree

File: _data/history.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  img: https://photos.newtheatre.org.uk/New-Theatre-Past/i-MMZNR6c/0/M/382108203_6cf1e5832e_o-M.jpg
+  alt-img: Old stage
```
Comment (wjdp) [inline _data/history.yaml]: img and alt should be under the same key, see:

```yaml
image:
  href: http...
  alt: Some descriptive alternate text
```

then 

```html
<img src="{{ image.href }}" alt="{{ image.alt }}" />
```

File: _layouts/history.html (hunk 4 above)
Comment (wjdp) [inline _layouts/history.html]: Eh, in the current state doesn't really need the extra work/overhead

File: _layouts/history.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+layout: page
+body_class: layout-page
+---
+
+{{ content }}
+<h1> History of the Nottingham New Theatre </h1>
+
+{% for node in site.data.history %}
```
Comment (wjdp) [inline _layouts/history.html]: Can this go in an include? `_includes/timeline.html`

File: _layouts/history.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (wjdp) [inline _layouts/history.html]: Liquid tags allowed in content pages, layout page is unnecessary.

File: _data/history.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+- year: 1940s
+  title: Formation of TEC
+  description: "Previously The University of Nottingham Technical Executive Committee, and now [TEC PA and Lighting](https://nottinghamtec.co.uk), this group had oversight of a number of departments across the university and Students' Union. The Stage Productions Department was responsible for lighting and sound in the student theatre. <br />The SPD Manager also served as Technical Director of Dramsoc, in addition to overseeing the electrics at the annual University Opera, performed in the Great Hall of Trent Building"
+- year: 1945
+  title: Earliest New Theatre Graduate
+  description: "The earliest graduate in our records. Think you can beat that? Just drop us a line."
+- year: 1958
```
Comment (wjdp) [inline _data/history.yaml]: Can we have line breaks between the years? Makes easier to parse.

File: _content/history.md [hunk 9]
```diff
@@ -1,43 +1,14 @@
...
-
-![Studio A](https://photos.newtheatre.org.uk/201314/Seating-Appeal-2013/i-zB8NQLT/0/M/35%20Matthew%20closing-M.jpg){: .right}{: .page-photo}
-
-In 2001 an extended foyer was added to the building, following a donation from an alumnus of the University.
-
-Until recently, the New Theatre was contained within the Archaeology and Classics building. The summer of 2012 saw an extensive redevelopment of the building housing the New Theatre. The Archaeology and Classics building was demolished from the site, leaving the New Theatre as its own freestanding building. Parts of the old building were retained, and repurposed as new rehearsal rooms, a studio space, storage rooms and a new constume cupboard. The foyer was also significantly extended.
-
-In 2012 the committee voted to alter the name of the theatre from "The New Theatre" to "The Nottingham New Theatre".
+With a long and rich history, the Nottingham New Theatre has had a steady stream of members through its doors across many departments for the last 70 years.
```
Comment (wjdp) [inline _content/history.md]: Stick `{% include timeline.html %}` here (see comments on layout)

**Comment by samozzy** [comment]
Think all the reviews have been addressed. Thank you both for the pointers. Went for a compromise on the two titles and went for just one, to match the feel of the About page.

**Comment by nhyty** [comment]
Would some of this information be able to be used for #53 ???

**Comment by samozzy** [comment]
Definitely. If an event tag was added in the data file another page could index it looking for these.

**Comment by nhyty** [comment]
@wjdp This ready to be merged in??

**Comment by wjdp** [comment]
Apologies for the PR intrusion, thought this would be easier than requesting changes. Can @nhyty and @samozzy sign off on this please?

**Comment by samozzy** [comment]
Looks great, thank you! Do we want to add the logo in the bit about the new logo?
I also think more images all round, but that can probably come outside of this textual PR.

**Comment by nhyty** [comment]
@samozzy I was thinking about putting more images in but we can do that later on. I think we should definatley show the old branding then the new branding.

**Comment by wjdp** [comment]
Would leave out at the moment. The page is rather bare, but my idea would be to do some styling (vertical timeline thingy) afterwards.

While working through the changes thought it might be nice to have images on some items and SVG/icony line-art for more abstract items.

**Comment by nhyty** [comment]
@wjdp i thin its fine as si so shall i merge it in will?

**Comment by wjdp** [comment]
@nhyty Just add an approved review first (makes pretty checkmarks :) (and useful data))

**Comment by samozzy** [comment]
When you talk about those graphics are you thinking something we make ourselves or find in the form of the icons currently on the site?

**Comment by nhyty** [comment]
@wjdp yeah, thats fne

**Comment by wjdp** [comment]
@samozzy Would probably be custom graphics. Can scour sites such as https://thenounproject.com/ and/or sketch raster images in Inkscape.

**Pull request merged**